server/src/main/java/org/elasticsearch/index/codec/tsdb/DocValuesForUtil.java

```diff
@@ -25,32 +29,70 @@ public DocValuesForUtil(int blockSize) {
         this.blockSize = blockSize;
     }
 
-    void encode(long[] in, int bitsPerValue, DataOutput out) throws IOException {
+    public static int roundBits(int bitsPerValue) {
+        if (bitsPerValue > 32 && bitsPerValue <= BITS_IN_FIVE_BYTES) {
+            return BITS_IN_FIVE_BYTES;
+        } else if (bitsPerValue > BITS_IN_FIVE_BYTES && bitsPerValue <= BITS_IN_SIX_BYTES) {
+            return BITS_IN_SIX_BYTES;
+        } else if (bitsPerValue > BITS_IN_SIX_BYTES && bitsPerValue <= BITS_IN_SEVEN_BYTES) {
+            return BITS_IN_SEVEN_BYTES;
+        }
+        return bitsPerValue;
+    }
+
+    void encode(long[] in, int bitsPerValue, final DataOutput out) throws IOException {
         if (bitsPerValue <= 24) { // these bpvs are handled efficiently by ForUtil
             forUtil.encode(in, bitsPerValue, out);
         } else if (bitsPerValue <= 32) {
             collapse32(in);
             for (int i = 0; i < blockSize / 2; ++i) {
                 out.writeLong(in[i]);
             }
+        } else if (bitsPerValue == BITS_IN_FIVE_BYTES || bitsPerValue == BITS_IN_SIX_BYTES || bitsPerValue == BITS_IN_SEVEN_BYTES) {
+            encodeFiveSixOrSevenBytesPerValue(in, bitsPerValue, out);
         } else {
+            assert bitsPerValue > 56 : "bitsPerValue must be greater than 56 but was [" + bitsPerValue + "]";
             for (long l : in) {
                 out.writeLong(l);
             }
         }
     }
 
-    void decode(int bitsPerValue, DataInput in, long[] out) throws IOException {
+    private void encodeFiveSixOrSevenBytesPerValue(long[] in, int bitsPerValue, final DataOutput out) throws IOException {
+        int bytesPerValue = bitsPerValue / Byte.SIZE;
+        int padding = Long.BYTES - bytesPerValue;
+        byte[] encoded = new byte[bytesPerValue * blockSize + padding];
```

Comment:
maybe store a byte[1024] as an instance member to avoid allocating a new byte[] every time?